src/main/java/com/intuit/graphql/orchestrator/schema/transform/KeyTransformerPostMerge.java

```diff
@@ -0,0 +1,43 @@
+package com.intuit.graphql.orchestrator.schema.transform;
+
+import com.intuit.graphql.graphQL.TypeDefinition;
+import com.intuit.graphql.orchestrator.federation.EntityTypeMerger;
+import com.intuit.graphql.orchestrator.federation.Federation2PureGraphQLUtil;
+import com.intuit.graphql.orchestrator.federation.extendsdirective.EntityExtension;
+import com.intuit.graphql.orchestrator.federation.extendsdirective.exceptions.BaseTypeNotFoundException;
+import com.intuit.graphql.orchestrator.xtext.XtextGraph;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class KeyTransformerPostMerge implements Transformer<XtextGraph, XtextGraph> {
```

Comment:
we do entity type merging here because a base type maybe processed by the xtext folder late